cinema/views.py

```diff
@@ -48,6 +76,28 @@ class MovieSessionViewSet(viewsets.ModelViewSet):
     queryset = MovieSession.objects.all()
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+
+        if self.action == "list":
+            queryset = queryset.annotate(
+                tickets_available=F("cinema_hall__rows")
+                * F("cinema_hall__seats_in_row")
+                - Count("tickets")
+            )
+
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+
+        if movie:
+            movie_id = str_to_int(movie)
+            queryset = queryset.filter(movie__id__in=movie_id)
```

Comment:
You could also extract filtering to a separate method here and in `MovieSession`.

Reply:
yes, that would be better

Reply:
but not a huge mistake. And in the same time, it isn't a best practice